src/update-feeds-loops/update-feeds-loops.ts

```diff
@@ -339,6 +339,10 @@ export const processBatch = async (
     }
   }
 
+  // Fetch the gas price regardless of whether there are any feeds to be updated or not in order for gas oracle to
+  // maintain historical gas prices.
+  await fetchAndStoreGasPrice(chainId, providerName, provider);
```

Comment:
Instead of here, isn't it better to call this function on `runUpdateFeeds` function? Because there is no meaningful time between processBatch calls and it will cause redundant RPC calls. For example, if we have 200 feeds on a chain and batch size 10, we probably requests same gas price 20 times for each RPC provider.

Reply:
The [batches are staggered](https://github.com/api3dao/airseeker-v2/blob/22a1ddae6585c2c8eb79f4f08ff36899e2cd2bb8/src/update-feeds-loops/update-feeds-loops.ts#L201) so there is a time delay. If we only fetched in in `runUpdateFeeds` the last batch would use gas price old ~`dataFeedUpdateInterval`.

 I see the point with multiple RPC calls though, but I don't see a better place when to fetch this.

Reply:
It makes sense. Your approach looks more proper 👍🏼